app/gui/src/dashboard/layouts/Drive/Categories/CategoriesProvider.tsx

```diff
@@ -1,35 +1,18 @@
-/**
- * @file
- *
- * Provider for the categories.
- */
-
-import { useEventCallback } from '#/hooks/eventCallbackHooks'
+/** @file Provider for categories. */
 import { useOffline } from '#/hooks/offlineHooks'
 import { useSearchParamsState } from '#/hooks/searchParamsStateHooks'
+import { driveLocationStore, getDriveLocation, setDriveLocation } from '#/providers/DriveProvider'
 import { useBackends, useFullUserSession } from '$/providers/react'
-import type { ReactNode } from 'react'
-import { isCloudCategory, type Category, type CategoryId } from './Category'
-import {
-  CategoriesContext,
-  categoryIdStore,
-  useCategories,
-  type CategoriesContextValue,
-} from './categoriesHooks'
+import { useEffect } from 'react'
+import { isCloudCategory, type CategoryId } from './Category'
+import { CategoriesContext, useCategories, type CategoriesContextValue } from './categoriesHooks'
 
-/**
- * Props for the {@link CategoriesProvider}.
- */
-export interface CategoriesProviderProps {
-  readonly children: ReactNode | ((contextValue: CategoriesContextValue) => ReactNode)
-  readonly onCategoryChange?: (previousCategory: Category | null, newCategory: Category) => void
-}
+/** Props for the {@link CategoriesProvider}. */
+export interface CategoriesProviderProps extends React.PropsWithChildren {}
 
-/**
- * Provider for the categories.
- */
-export function CategoriesProvider(props: CategoriesProviderProps): React.JSX.Element {
-  const { children, onCategoryChange = () => {} } = props
+/** Provider for categories. */
+export function CategoriesProvider(props: CategoriesProviderProps) {
+  const { children } = props
```

Comment:
I would elaborate more about responsibility of this provider. It - obviously - provides categories, but does not manage current category -- while _also_ synchronizes the URL param with Drive Store, which maybe violate the "single responsibility rule" a bit?